addons/project/models/project.py

```diff
@@ -2484,16 +2484,36 @@ def action_assign_to_me(self):
     def action_unassign_me(self):
         self.write({'user_ids': [Command.unlink(self.env.uid)]})
 
-    # If depth == 1, return only direct children
-    # If depth == 3, return children to third generation
-    # If depth <= 0, return all children without depth limit
     def _get_all_subtasks(self, depth=0):
-        children = self.mapped('child_ids')
-        if not children:
-            return self.env['project.task']
-        if depth == 1:
-            return children
-        return children + children._get_all_subtasks(depth - 1)
```

Comment:
I think we should do this code when `not any(this._ids)` is truly, to increase/decrease task count in the stat button when we add/remove subtask. 🙂